compiler/codegen/OMRTreeEvaluator.hpp

```diff
@@ -68,6 +70,24 @@ class TreeEvaluator
    static void evaluateNodesWithFutureUses(TR::Node *node, TR::CodeGenerator *cg);
    static void initializeStrictlyFutureUseCounts(TR::Node *node, vcount_t visitCount, TR::CodeGenerator *cg);
 
+   /**
+    * Check whether memory barriers must be created for a given memory operation.
+    *
+    * To avoid potential race conditions on SMP, memory barriers must sometimes be created.
+    * These usually take the form of sync instructions.
+    *
+    * One case where sync instructions are required is when accessing volatile memroy.
+    * Another case is when doing atomic memory operations on shadow references
+    * (indirect memory references; e.g. de-referencing an array element by index) as
+    * maintaining the ordering of the operations is required for them to be atomic.
+    */
```

Comment:
Is this function only used on Power?  If so, should it reside in the P specialized TreeEvaluator?  If not, which I suspect is the case, then perhaps it should be name more generically, like `isWriteMemoryBarrierNeeded`